runners/spark/src/main/java/org/apache/beam/runners/spark/SparkPipelineRunner.java

```diff
@@ -123,10 +135,66 @@ public PortablePipelineResult run(RunnerApi.Pipeline pipeline, JobInfo jobInfo)
         "Will stage {} files. (Enable logging at DEBUG level to see which files will be staged.)",
         pipelineOptions.getFilesToStage().size());
     LOG.debug("Staging files: {}", pipelineOptions.getFilesToStage());
-
     PortablePipelineResult result;
     final JavaSparkContext jsc = SparkContextFactory.getSparkContext(pipelineOptions);
 
+    EventLoggingListener eventLoggingListener = null;
+    if (pipelineOptions.getEventLogEnabled()) {
+      eventLoggingListener =
+          new EventLoggingListener(
+              jobInfo.jobId(),
+              new scala.Option<String>() {
+                @Override
+                public boolean isEmpty() {
+                  return false;
+                }
+
+                @Override
+                public String get() {
+                  return jobInfo.jobName();
+                }
+
+                @Override
+                public Object productElement(int i) {
+                  return null;
+                }
+
+                @Override
+                public int productArity() {
+                  return 0;
+                }
+
+                @Override
+                public boolean canEqual(Object o) {
+                  return false;
+                }
+              },
+              new URI(pipelineOptions.getSparkHistoryDir()),
+              jsc.getConf(),
+              jsc.hadoopConfiguration());
+      eventLoggingListener.initializeLogIfNecessary(false, false);
+      eventLoggingListener.start();
+      scala.collection.immutable.Map<String, String> logUrlMap =
+          new scala.collection.immutable.HashMap<String, String>();
+      Tuple2<String, String>[] sparkConfList = jsc.getConf().getAll();
+      String sparkMaster = "";
+      String sparkExecutorID = "";
+      for (Tuple2<String, String> sparkConf : sparkConfList) {
+        if (sparkConf._1().equals("spark.master")) {
+          sparkMaster = sparkConf._2();
+        } else if (sparkConf._1().equals("spark.executor.id")) {
+          if (!sparkExecutorID.equals(sparkConf._2())) {
```

Comment:
The current logic is, "if the Spark executor id is not the same as the last executor id we saw, log it." It may be better to just log every Spark executor id. I couldn't find any documentation on `spark.executor.id`, but I assume there wouldn't be duplicates. And even if there are duplicates, I don't think logging them will harm anything.

Reply:
Ok, I've changed this along with `getAllWithPrefix("spark.executor.id")`.